rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -150,28 +150,44 @@ std::vector<rsx_subresource_layout> get_subresources_layout(const rsx::texture &
 	bool is_swizzled = !(texture.format() & CELL_GCM_TEXTURE_LN);
 	switch (format)
 	{
-	case CELL_GCM_TEXTURE_D8R8G8B8:
-	case CELL_GCM_TEXTURE_A8R8G8B8:
-		return get_subresources_layout_impl<1, u32>(pixels, w, h, depth, layer, texture.get_exact_mipmap_count(), texture.pitch(), !is_swizzled);
+	case CELL_GCM_TEXTURE_B8:
+		return get_subresources_layout_impl<1, u8>(pixels, w, h, depth, layer, texture.get_exact_mipmap_count(), texture.pitch(), !is_swizzled);
+
+	case ~(CELL_GCM_TEXTURE_LN | CELL_GCM_TEXTURE_UN) & CELL_GCM_TEXTURE_COMPRESSED_B8R8_G8R8:
+	case ~(CELL_GCM_TEXTURE_LN | CELL_GCM_TEXTURE_UN) & CELL_GCM_TEXTURE_COMPRESSED_R8B8_R8G8:
+	case CELL_GCM_TEXTURE_COMPRESSED_HILO8:
+	case CELL_GCM_TEXTURE_COMPRESSED_HILO_S8:
 	case CELL_GCM_TEXTURE_DEPTH16:
+	case CELL_GCM_TEXTURE_DEPTH16_FLOAT: // Untested
```

Comment:
I only needed to add that, so the texture formats test would boot. It and other untested formats should work, but they should be verified. I could probably go through the trouble of updating the tests and the texture format generation script to also add support for generation of such formats. I'll look into it.